packages/engine/src/worker/runner/javascript/hash_util.js

```diff
@@ -77,6 +112,10 @@
       const child = children[j];
       const name = child.name;
       obj[name] = _vector_to_array(struct[name]);
+      if (child.type.listSize) {
```

Comment:
I also don't understand why this is here? Should the logic not be in `_vector_to_array`?